### DISCUSSION TITLE
Creating a series of snap tracks

### DISCUSSION THREAD

**Discussion by superm1** [fwupd/fwupd#5576] [General]
As for the next release of snapd and 0.8.11 release of fwupd will support using it confined.

I think we're a lot closer to being able to advertise its regular use once we fix for a few remaining bugs: (https://github.com/fwupd/fwupd/issues/5484, https://github.com/fwupd/fwupd/issues/5542)

As part of this, I think it makes sense to also adopt the concept of [tracks](https://ubuntu.com/blog/controlling-snap-releases-with-channels-tracks-and-branches-part-2) for the snap starting with our first 0.9.x release.

The idea would be that our `main` branch always tracks the `latest` track and `0.8.x` branch tracks the `0.8.x` track.

Here's my proposal:

_main_
- We don't turn off snap publishing when we fork for "stable" branch releases.
- CI builds from `main` would publish to `latest/edge`.
- Tagged builds `main` would promote to `latest/candidate`.
- 7 days after a release we would promote `main` builds from `latest/candidate` to `latest/stable`

_0.8.x_
- CI builds from `0.8.x` would publish to `0.8.x/edge`
- Tagged builds from `0.8.x` would promote to `0.8.x/candidate`
- 7 days after a release we would promote `0.8.x` builds from `0.8.x/candidate` to `0.8.x/stable`

That is - whenever we branch a release we would make a new track, and follow this thought process.

@xnox, @valentindavid @hughsie @kenvandine thoughts please?

**Comment by hughsie** [comment]
Seems sane to me, although I'm not really familiar with snap tracks. From a 40,000ft view it does seem a little hard to explain.

**Reply by superm1** [reply]
I think from a 40k foot view you can look at it as track latest fwupd release vs track " stable " release.

Think like you have a regression in 1.9.3 and can swap to 1.8.13 instead and keep tracking 1.8.x releases until you're confident it's fixed in 1.9.x.
